File: 1-TwoSum/1-TwoSum.java
// Last updated: 5/24/2025, 1:51:14 AM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> numMap = new HashMap<>();
       int n = nums.length;
       for(int i = 0;i < n;i++){
        int complement = target - nums[i];
        if(numMap.containsKey(complement) ){
            return new int[]{i, numMap.get(complement)};
        }
        numMap.put(nums[i], i);
       }

       return new int[]{};
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 6/15/2025, 12:24:48 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       unordered_map<int, int> hash;
        for(int i=0; i < nums.size();++i){
       int complement = target - nums[i];
            if(hash.find(complement)!= hash.end()){
                return {hash[complement],i};
            }
            hash[nums[i]] = i;            
        }
        return{};
    }
};